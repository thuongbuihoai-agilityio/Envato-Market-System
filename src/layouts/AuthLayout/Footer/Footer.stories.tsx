import { StoryObj, Meta } from '@storybook/react';

// components
import Footer from '@app/layouts/AuthLayout/Footer';

const meta: Meta<typeof Footer> = {
  title: 'Footer Form',
  tags: ['autodocs'],
  component: Footer,
};

type Story = StoryObj<typeof Footer>;

export const Default: Story = {};

export default meta;
