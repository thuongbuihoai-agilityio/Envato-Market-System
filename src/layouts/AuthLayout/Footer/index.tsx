import { Box, Text } from '@chakra-ui/react';

// utils
import { getCurrentYear } from '@app/utils/time';

const Footer = () => {
  const currentYear = getCurrentYear();

  return (
    <Box mt={24}>
      <Text
        mt={6}
        fontSize="sm"
        variant="textSm"
        textAlign="center"
        _dark={{ color: 'secondary.700' }}
      >
        @ {currentYear} Bankco. All Rights Reserved.
      </Text>
    </Box>
  );
};

export default Footer;
