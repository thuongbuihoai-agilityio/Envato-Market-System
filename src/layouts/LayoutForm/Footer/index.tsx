import { Box, Flex, Text, Link, List, ListItem } from '@chakra-ui/react';

const Footer = () => (
  <Box p={4} textAlign="center">
    <Flex
      justifyContent="center"
      alignItems="center"
      flexWrap="wrap"
      columnGap="40px"
      fontSize="sm"
      _dark={{ color: 'secondary.100' }}
      _light={{ color: 'gray.600' }}
      as={List}
    >
      <ListItem mx={2}>
        <Link href="#" _hover={{ color: 'gray.500' }}>
          Terms & Conditions
        </Link>
      </ListItem>
      <ListItem mx={2}>
        <Link href="#" _hover={{ color: 'gray.500' }}>
          Privacy Policy
        </Link>
      </ListItem>
      <ListItem mx={2}>
        <Link href="#" _hover={{ color: 'gray.500' }}>
          Help
        </Link>
      </ListItem>
      <ListItem mx={2}>
        <Link href="#" _hover={{ color: 'gray.500' }}>
          English
        </Link>
      </ListItem>
    </Flex>
    <Text
      mt={4}
      fontSize="sm"
      variant="textSm"
      _dark={{ color: 'secondary.700' }}
    >
      @ 2023 Bankco. All Rights Reserved.
    </Text>
  </Box>
);

export default Footer;
