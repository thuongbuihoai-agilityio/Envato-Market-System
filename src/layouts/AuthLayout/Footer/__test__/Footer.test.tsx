import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

// Components
import Footer from '@app/layouts/AuthLayout/Footer';

jest.mock('@app/utils/time', () => ({
  getCurrentYear: jest.fn(() => 2023),
}));

test('renders Footer with links and current year', () => {
  const { container } = render(<Footer />);

  expect(container).toMatchSnapshot();
});
