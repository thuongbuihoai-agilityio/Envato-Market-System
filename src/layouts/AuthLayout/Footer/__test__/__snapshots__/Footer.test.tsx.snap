// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer with links and current year 1`] = `
<div>
  <div
    class="css-178yklu"
  >
    <ul
      class="css-1mzm5ub"
      role="list"
    >
      <li
        class="css-z9iakq"
      >
        <a
          class="chakra-link css-1bl9y41"
          href="#"
        >
          Terms & Conditions
        </a>
      </li>
      <li
        class="css-z9iakq"
      >
        <a
          class="chakra-link css-1bl9y41"
          href="#"
        >
          Privacy Policy
        </a>
      </li>
      <li
        class="css-z9iakq"
      >
        <a
          class="chakra-link css-1bl9y41"
          href="#"
        >
          Help
        </a>
      </li>
      <li
        class="css-z9iakq"
      >
        <a
          class="chakra-link css-1bl9y41"
          href="#"
        >
          English
        </a>
      </li>
    </ul>
    <p
      class="chakra-text css-e9tuiy"
    >
      @ 
      2023
       Bankco. All Rights Reserved.
    </p>
  </div>
</div>
`;
